### DISCUSSION TITLE
Passing a List[FlyteFile] to a raw ContainerTask

### DISCUSSION THREAD

**Discussion by djbutler** [flyteorg/flyte#4282] [Q&A]
Is it possible to pass a list of FlyteFiles to a raw ContainerTask?

For example:

```
concat_files = ContainerTask(
    name="concat-files",
    input_data_dir="/var/inputs",
    output_data_dir="/var/outputs",
    inputs=kwtypes(infiles=List[FlyteFile])
    outputs=kwtypes(outfile=FlyteFile),
    image="alpine",
    command=[
        "cat",
        "{{.inputs.infiles[0]}}",
        "/var/outputs/outfile",
    ],
)
```

This doesn't seem to work for me. If this is possible at all, I'm guessing this syntax is incorrect: `{{.inputs.infiles[0]}}`

Any tips?

**Comment by Future-Outlier** [comment]
Hi, @djbutler .
I think the answer will be `"{{.inputs.infiles.0}}"`
and the command should be 
```python
command=[
        "cat",
        ">",
        "{{.inputs.infiles.0}}",
        "/var/outputs/outfile",
    ],
```

**Reply by Future-Outlier** [reply]
Here are the references, thank you!
https://github.com/flyteorg/flyte/blob/master/flytecopilot/data/download.go#L251-L267

**Reply by Future-Outlier** [reply]
Wait, I think currently raw container task doesn't support collection, I will create an issue for it.

**Reply by Future-Outlier** [reply]
The implementation has bug.

**Reply by Future-Outlier** [reply]
I think the dictionary might has bug too.

**Comment by Future-Outlier** [comment]
I think its not possible to use `template` syntax to do it, but flyte directory is almost done, thank you.

**Comment by Future-Outlier** [comment]
flytedirectory is supported as input and output now!
